### PR TITLE
feat: add compatible Aliyun interface

### PR DESCRIPTION
**Pull request by xiaohan1202**

### CODE REVIEW

File: controllers/cluster/go.mod [hunk 1]
```diff
@@ -22,19 +22,6 @@ require (
 	github.com/Azure/go-autorest/autorest/date v0.3.0 // indirect
 	github.com/Azure/go-autorest/logger v0.2.1 // indirect
 	github.com/Azure/go-autorest/tracing v0.6.0 // indirect
-	github.com/aws/aws-sdk-go-v2 v1.16.16 // indirect
```
Comment (zzjin) [inline controllers/cluster/go.mod:25]: should run `go mod tidy` to update modules

File: controllers/infra/drivers/aws/create_instance.go [hunk 2]
```diff
@@ -172,7 +172,7 @@ func (d Driver) GetTags(hosts *v1.Hosts, infra *v1.Infra) []types.Tag {
 // get instace data by id
 func getInstancesByID(id *string, client *ec2.Client) ([]types.Reservation, error) {
 	uidKey := fmt.Sprintf("tag:%s", common.InfraInstancesUUID)
-	statusName := common.InstanceState
+	statusName := "instance-state-name"
```
Comment (zzjin) [inline controllers/infra/drivers/aws/create_instance.go:175]: "instance-state-name" const should still be const, you can rename it to words like AWSInstanceState

File: controllers/infra/common/common.go [hunk 3]
```diff
@@ -15,10 +15,9 @@ const (
 	TrySleepTime         = time.Second
 	TRUELable            = "true"
 	SealosInfraFinalizer = "infra.sealos.io/finalizers"
-	InfraSecretPrefix    = "infra-secret"
 	InfraVolumeIndex     = "infra.sealos.io/volumes/index"
 	VolumeInfraID        = "infra.sealos.io/volumes/infraID"
-	InstanceState        = "instance-state-name"
+	InfraSecretPrefix    = "infra-secret"
```
Comment (fanux) [inline controllers/infra/common/common.go:20]: "sealos.io/infra/secret"

File: controllers/infra/controllers/infra_controller.go [hunk 4]
```diff
@@ -76,6 +81,16 @@ func (r *InfraReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	if err := r.Get(ctx, req.NamespacedName, infra); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
+
+	// choose driver
+	platform := infra.Spec.Platform
```
Comment (fanux) [inline controllers/infra/controllers/infra_controller.go:86]: Put those logic into NewDriver function.

File: controllers/infra/drivers/driver.go [hunk 5]
```diff
@@ -65,3 +69,11 @@ func NewDriver() (Driver, error) {
 		Client: client,
 	}, nil
 }
+
```
Comment (fanux) [inline controllers/infra/drivers/driver.go:72]: Also nees `NewDriver(platform string)(Driver, error)` function.

File: controllers/infra/common/common.go (hunk 3 above)
Comment (xiaohan1202) [reply controllers/infra/common/common.go:20]: > 

this name is invalid in k8s

**Pull request closed without merging**